edx_proctoring/static/proctoring/js/views/proctored_exam_view.js

```diff
@@ -0,0 +1,141 @@
+            var taking_as_proctored = this.model.get('taking_as_proctored');
+            var time_left = this.model.get('time_remaining_seconds') > 0;
+            var status = this.model.get('attempt_status');
+            var in_courseware = document.location.href.indexOf('/courses/' + this.model.get('course_id') + '/courseware/') > -1;
+
+            if ( taking_as_proctored && time_left && in_courseware && status !== 'started'){
+                $(window).bind('beforeunload', this.unloadMessage);
+            } else {
+                // remove callback on unload event
+                $(window).unbind('beforeunload', this.unloadMessage);
+            }
+
+            this.render();
+        },
+        render: function () {
+            if (this.template !== null) {
+                if (
+                    this.model.get('in_timed_exam') &&
+                    this.model.get('time_remaining_seconds') > 0 &&
+                    this.model.get('attempt_status') !== 'error'
+                ) {
+                    var html = this.template(this.model.toJSON());
+                    this.$el.html(html);
+                    this.$el.show();
+                    this.updateRemainingTime(this);
+                    this.timerId = setInterval(this.updateRemainingTime, 1000, this);
+
+                    // put the exam controls (namely stop button)
+                    // right after the sequence naviation ribbon
+                    html = this.controls_template(this.model.toJSON());
+                    $('.sequence-nav').after(html);
+
+                    // Bind a click handler to the exam controls
+                    var self = this;
+                    $('.proctored-exam-action-stop').click(function(){
+                        $(window).unbind('beforeunload', self.unloadMessage);
+
+                        $.ajax({
+                            url: '/api/edx_proctoring/v1/proctored_exam/attempt/' + self.model.get('attempt_id'),
+                            type: 'PUT',
+                            data: {
+                              action: 'stop'
+                            },
+                            success: function() {
+                              // Reloading page will reflect the new state of the attempt
+                              location.reload();
+                            }
+                        });
+                    });
+                    $('.proctored-exam-action-stop').css('cursor', 'pointer');
+                }
+            }
+            return this;
+        },
+        unloadMessage: function  () {
+            return null;
+            return gettext("As you are currently taking a proctored exam,\n" +
+                "you should not be navigation away from the exam.\n" +
+                "This may be considered as a violation of the \n" +
+                "proctored exam and you may be disqualified for \n" +
+                "credit eligibility in this course.\n");
+        },
```

Comment:
@griffresch What message do we want here? I believe students can navigate away and go elsewhere (in the course?). The timer continues the countdown. Students can return to the exam and resume at the same point within the exam, but they have lost whatever time they spent away.